### DISCUSSION TITLE
Qwen2推理加速没有效果（accelerate、flash_attention_2、deepspeed）

### DISCUSSION THREAD

**Discussion by kofuya** [QwenLM/Qwen3#1077] [Q&A]
def __init__(self):
        accelerator = Accelerator()
        self.device = accelerator.device
        self.model = AutoModelForCausalLM.from_pretrained(
            "qwen/Qwen2-7B-Instruct",
            torch_dtype=torch.bfloat16,
            device_map={"": accelerator.process_index},
            attn_implementation="flash_attention_2"
        )
        self.tokenizer = AutoTokenizer.from_pretrained(
            model_path+"qwen/Qwen2-7B-Instruct")
 无论是否启用了accelerate、flash_attention_2、deepspeed，显存占用、推理耗时都没变化，测验了短文本、长文本都是相同的情况，这是为什么？？

**Comment by jklj077** [comment]
https://github.com/QwenLM/Qwen2.5/issues/1078
